modules/nodesale/internal/validator/delegate/validator.go

```diff
@@ -0,0 +1,52 @@
+package delegate
+
+import (
+	"context"
+
+	"github.com/cockroachdb/errors"
+	"github.com/gaze-network/indexer-network/modules/nodesale/datagateway"
+	"github.com/gaze-network/indexer-network/modules/nodesale/internal/entity"
+	"github.com/gaze-network/indexer-network/modules/nodesale/internal/validator"
+	"github.com/gaze-network/indexer-network/modules/nodesale/protobuf"
+	"github.com/samber/lo"
+)
+
+type DelegateValidator struct {
```

Comment:
Is there a reason to split `DelegateValidator` and `PurchaseValidator` from `Validator`? Should probably merge all of this to one `Validator` so package name is not separated.

Reply:
They use different check. I would forget which validator is for which part.